cmd/apps/rancher_app.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) arkade author(s) 2020. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package apps
+
+import (
+	"fmt"
+	"github.com/alexellis/arkade/pkg"
+	"github.com/spf13/cobra"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path"
+	"strings"
+)
+
+func MakeInstallRancher() *cobra.Command {
+	var command = &cobra.Command{
+		Use:          "rancher",
+		Short:        "Install rancher to import and manage Kubernetes",
+		Long:         `Install rancher to import and manage Kubernetes clsuter.`,
+		Example:      `  arkade install rancher`,
+		SilenceUsage: true,
+	}
+
+	command.RunE = func(command *cobra.Command, args []string) error {
+		kubeConfigPath := getDefaultKubeconfig()
+
+		if command.Flags().Changed("kubeconfig") {
+			kubeConfigPath, _ = command.Flags().GetString("kubeconfig")
+		}
+
+		fmt.Printf("Using kubeconfig: %s\n", kubeConfigPath)
+
+		arch := getNodeArchitecture()
+		fmt.Printf("Node architecture: %q\n", arch)
+
+		if arch != IntelArch && arch != "arm" {
+			return fmt.Errorf(`only Intel and "arm" is supported for this app`)
+		}
+
+		_, err := kubectlTask("create", "ns",
+			"cattle-system")
+		if err != nil {
+			if !strings.Contains(err.Error(), "exists") {
+				return err
+			}
+		}
+
+		req, err := http.NewRequest(http.MethodGet,
```

Comment:
Not comfortable with using your private GitHub for this. How are folks supposed to install Rancher? Does it not have helm? cc @ibuildthecloud

Reply:
there is a helm chart as well for installing.

Reply:
@saiyam1814 Can we use the helm chart?  For the cert-manager install arkade already has cert-manager, so can you just have arkade install that first? I think the only crappy part of the helm install is that it requires a DNS domain, right?

Reply:
Hi @alexellis and @Ibuildthecloud  I did ran the helm chart on a fresh kubernetes cluster. 
So it requires 2 things heml3, cert-manager install , and DNS domain. 
Cert-manager we can directly install via arkade  but hostname is something that make the installation hard. How to deal with that? Right now the current yaml that I have in the code simple installs rancher deployment(without TLS) and imports the local cluster. 
Also there are so many options that can be provided in the helm chart for rancher @alexellis do we need all the options ?

Reply:
here is the ingress that gets created:
`apiVersion: v1
items:
- apiVersion: extensions/v1beta1
  kind: Ingress
  metadata:
    annotations:
      cert-manager.io/issuer: rancher
      nginx.ingress.kubernetes.io/proxy-connect-timeout: "30"
      nginx.ingress.kubernetes.io/proxy-read-timeout: "1800"
      nginx.ingress.kubernetes.io/proxy-send-timeout: "1800"
    creationTimestamp: "2020-04-03T09:48:59Z"
    generation: 1
    labels:
      app: rancher
      chart: rancher-2.4.2
      heritage: Helm
      release: rancher
    name: rancher
    namespace: cattle-system
    resourceVersion: "1737"
    selfLink: /apis/extensions/v1beta1/namespaces/cattle-system/ingresses/rancher
    uid: 92400c93-d2e0-4f0a-8e0a-d83cccf70105
  spec:
    rules:
    - host: localhost
      http:
        paths:
        - backend:
            serviceName: rancher
            servicePort: 80
    tls:
    - hosts:
      - localhost
      secretName: tls-rancher-ingress
  status:
    loadBalancer: {}
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""`

Another thing is in civo marketplace where I have the rancher app installation it also have ingress but. like this :
`
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: cattle-ingress
  namespace: cattle-system
  annotations:
    kubernetes.io/ingress.class: traefik
spec:
  rules:
  - http:
      paths:
      - path: /
        backend:
          serviceName: cattle-service
          servicePort: http
`